src/constants.ts

```diff
@@ -18,15 +18,15 @@ export const PRIORITY = {
   LOW: 2,
 };
 
-export const PHASE_LAZY_DELAY = 50;
-
 export const COLLECTED = new Map();
 
-export const SETTINGS: Settings = {
+export const DEFAULT_SETTINGS: Settings = {
   CURRENT_MODE: MODE.HYDRATE,
   MANIFEST: {
     publicPath: '/',
     assets: {},
   },
   CROSS_ORIGIN: undefined,
 };
+
+export const SETTINGS: Settings = Object.assign({}, DEFAULT_SETTINGS);
```

Comment:
It's a problem outside of tests as well, provided consumers call `init` multiple times for whatever reason. The crux of it is that calling `init` with particular options, may not be applied correctly if `SETTINGS` is already set.

This would be less of a problem if we didn't have a mutable global, and instead converted `init` to a context provider. Though I presume you wouldn't be a fan of that, and technically the SSR HTML collect function would occur a bit later.

How do you see this working with getters and setters?

Reply:
Something like
```js
const defaultSettings = { ... };
let settings = {};

export const getConfig = (key) => {
  return settings[key] ?? defaultSettings[key]
}

export const setConfig = (key, value) => {
  settings[key] = value
}

export const resetConfig = () => {
  settings = {}
}
```

Reply:
Thanks, I ended up simplifying this a bit so that it's easier to read and ensured it is type safe.